src/front/glsl/ast.rs

```diff
@@ -134,19 +134,117 @@ pub enum HirExprKind {
     },
 }
 
+#[derive(Debug, Hash, PartialEq, Eq)]
+pub enum QualifierKey {
+    String(String),
+    Layout,
```

Comment:
I'm curious why the layout is special here?

Reply:
`Layout` applies both to `std140` and `std430`, so you can have
```glsl
layout(std140, std430, std140)
```
and you'll get only the final `std140` (this is horrible but that is how glsl defines it)

Reply:
maybe adding a short comment about this would help :)